pydantic/config.py

```diff
@@ -182,6 +215,7 @@ def get_config(config: ConfigDict | dict[str, Any] | type[Any] | None) -> Config
         )
         config_dict = {k: getattr(config, k) for k in dir(config) if not k.startswith('__')}
 
+    prepare_config(config_dict, error_label or 'ConfigDict')
```

Comment:
I added this due to some inconsistency in when invalid values of `extra` would result in more useful error messages.

Before, you would get the more explicit error for things that created models, but not for dataclasses. In principle I think type hints should prevent issues, but I think this is an improvement because using `"ignore"` instead of `Extra.ignore` is a pretty easy mistake to make considering it was okay in v1, and if you do that, you get a confusing (in my opinion) error saying something like `type 'str' has no attribute 'value'` (since it tries to grab the .value off the enum).